pkg/udevevent/event.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2018 The OpenEBS Author
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package udevevent
+
+/*
+#cgo LDFLAGS: -ludev
+#include <libudev.h>
+*/
+import "C"
+import (
+	"syscall"
+
+	"github.com/openebs/node-disk-manager/pkg/util"
+
+	"github.com/golang/glog"
+	"github.com/openebs/node-disk-manager/cmd/controller"
+)
+
+//MonitorEvent start monitoring on udev source.
+func MonitorEvent() {
+	var udev *C.struct_udev
+	udev = C.udev_new()
+	if udev == nil {
+		glog.Fatal("Error while creating new udev object")
+	}
+	monitorDevices(udev)
+	C.udev_unref(udev)
+}
+
+func processDevice(dev *C.struct_udev_device) {
+	if dev != nil {
```

Comment:
Will it be good if this function returns error?

Reply:
Can we rewrite this function in a way that avoids nested logic?

Reply:
refactored that code avoid nested logic.